test/unit/document/document_test.ts

```diff
@@ -330,4 +330,63 @@ describe('DocumentReplica', function () {
       paths.push('$.rich');
     });
   });
+
+  it('insert elements before a specific node of array', function () {
+    const doc = DocumentReplica.create('test-col', 'test-doc');
+    doc.update((root) => {
+      root.data = [0, 1, 2];
+    });
+    assert.equal('{"data":[0,1,2]}', doc.toSortedJSON());
+    assert.equal(3, doc.getRoot().data.length);
+
+    doc.update((root) => {
+      const zero = root.data.getElementByIndex(0);
+      root.data.insertBefore(zero.getID(), 3);
+    });
+    assert.equal('{"data":[3,0,1,2]}', doc.toSortedJSON());
+    assert.equal(4, doc.getRoot().data.length);
+
+    doc.update((root) => {
+      const one = root.data.getElementByIndex(2);
+      root.data.insertBefore(one.getID(), 4);
+    });
+    assert.equal('{"data":[3,0,4,1,2]}', doc.toSortedJSON());
+    assert.equal(5, doc.getRoot().data.length);
+
+    doc.update((root) => {
+      const two = root.data.getElementByIndex(4);
+      root.data.insertBefore(two.getID(), 5);
+    });
+    assert.equal('{"data":[3,0,4,1,5,2]}', doc.toSortedJSON());
+    assert.equal(6, doc.getRoot().data.length);
+  });
+
+  it('insert elements before a deleted node of array', function () {
```

Comment:
How about moving this test to `array_test.ts`? I'm curious the reason why this test was added 🤔 
And I think it would be great to change test title like `can insert an element before specific position after delete operation ` etc.